src/main/java/com/alrex/parcool/api/compatibility/AbstractClientPlayerWrapper.java

```diff
@@ -0,0 +1,26 @@
+package com.alrex.parcool.api.compatibility;
+
+import net.minecraft.client.entity.player.AbstractClientPlayerEntity;
+import net.minecraftforge.event.TickEvent.PlayerTickEvent;
+
+public class AbstractClientPlayerWrapper extends PlayerWrapper {
+    private AbstractClientPlayerEntity player;
+    private static final WeakCache<AbstractClientPlayerEntity, AbstractClientPlayerWrapper> cache = new WeakCache<>();
```

Comment:
Though not limited to this class, `WeakCache` class of this static field contains `java.util.WeakHashMap` having `AbstractClientPlayerEntity` as key and `AbstractClientPlayerWrapper` itself as value right? 
But `AbstractClientPlayerWrapper` has `AbstractClientPlayerEntity` instance as strong reference. If I remember correctly `WeakHashMap`'s keys are weak reference but the values are strong reference. So in this case the keys which are weak reference are strongly referenced by their own values. The values are strongly referenced by `WeakHashMap` itself. As a result I think GC cannot collect the key objects?

Reply:
I think you're correct. To fix that I need to maintain a weak reference for the entity itself inside the wrapper, thank you for pointing that out